src/dymoprint/gui/q_dymo_labels_list.py

```diff
@@ -79,48 +105,98 @@ def dropEvent(self, e) -> None:
 
     def update_params(
         self,
+        dymo_labeler: DymoLabeler,
+        h_margin_mm: float,
+        min_label_width_mm: float,
         render_context: RenderContext,
-        min_payload_len_px: int,
-        justify="center",
+        justify: str = "center",
     ):
         """Update the render context used for rendering the label.
 
         Args:
         ----
-            justify: justification [center,left,right]
-            min_payload_len_px: minimum payload size in pixels
+            dymo_labeler: an instance of DymoLabeler object
+            h_margin_mm: horizontal margin [mm]
+            min_label_width_mm: minimum label width [mm]
             render_context (RenderContext): The new render context to use.
+            justify: justification [center,left,right]
         """
-        self.min_payload_len_px = min_payload_len_px
+        self.dymo_labeler = dymo_labeler
+        self.h_margin_mm = h_margin_mm
+        self.min_label_width_mm = min_label_width_mm
         self.justify = justify
         self.render_context = render_context
         for i in range(self.count()):
             item_widget = self.itemWidget(self.item(i))
             item_widget.render_context = render_context
         self.render_label()
 
-    def render_label(self):
-        """Render the label using the current render context and emit renderSignal."""
+    @property
+    def _payload_render_engine(self):
         render_engines = []
         for i in range(self.count()):
             item = self.item(i)
             item_widget = self.itemWidget(self.item(i))
             if item_widget and item:
                 item.setSizeHint(item_widget.sizeHint())
                 render_engines.append(item_widget.render_engine)
-        render_engine = HorizontallyCombinedRenderEngine(
-            render_engines=render_engines,
-            min_payload_len_px=self.min_payload_len_px,
-            max_payload_len_px=None,
+        return HorizontallyCombinedRenderEngine(render_engines=render_engines)
+
+    def _render_with_margins(self, mode: MarginsMode, signal: QtCore.pyqtSignal):
+        """Render the label using the current render context and emit renderSignal."""
+        render_engine = PrintPreviewRenderEngine(
+            render_engine=self._payload_render_engine,
+            justify=self.justify,
+            visible_horizontal_margin_px=mm_to_px(self.h_margin_mm),
+            labeler_margin_px=DymoLabeler.get_labeler_margin_px(),
+            max_width_px=None,
+            min_width_px=mm_to_px(self.min_label_width_mm),
+        )
+        try:
+            bitmap = render_engine.render(self.render_context)
+        except BaseException as err:  # noqa: BLE001
```

Comment:
I'd strongly recommend against introducing this and the next blind except a few lines down. They make things extremely difficult to maintain down the line.

Even if it's a generic catch-all thing, at least make an effort to catch all the common errors and leave `BaseException` as a failsafe rather than a primary measure.